src/Workspaces/CSharp/Portable/Extensions/SemanticModelExtensions.cs

```diff
@@ -237,6 +232,10 @@ private static bool CanBindToken(SyntaxToken token)
                 {
                     current = ((CastExpressionSyntax)current).Expression;
                 }
+                else if (current is DeclarationExpressionSyntax)
```

Comment:
Possible, but it would be inconsistent with the surrounding code. Probably this whole method could be refactored as a switch on Kind, but that is out of scope for this PR.